dace/transformation/interstate/sdfg_nesting.py

```diff
@@ -541,6 +541,10 @@ def apply(self, sdfg: SDFG):
                     '(reconnecting inputs)')
             state.add_edge(edge.src, edge.src_conn, node, edge.dst_conn,
                            edge.data)
+            # Fission state if necessary
```

Comment:
what happens if the inlined SDFG is inside a scope (e.g., two connected components in a map)? would it automatically not create two WCCs?

Also, what if there were already more than one WCC in the graph before inlining? Would it fission states that shouldn't be?

Reply:
This detects the case where a weakly connected group of nodes in the parent state (A) that was connected to the nested SDFG ends up disconnected from the subgraph of the nested SDFG (B) after inlining. This may create a data race since (A) may be executed in parallel with (B) or in the wrong order after the transformation. The solution is to move (A) to a separate state before (after) if (A) was input (output) to the nest SDFG.

Your first question may be a real issue. Therefore, the current check may fail in some cases. It might be better to check that none of the nodes in (B) are part of (A).

I do not think that this is an issue. If there is a group (A) detected, it will be moved in a separate state from the rest of the nodes, regardless of how may other WCCs exist in the state (after the transformation).